src/crimson/osd/osd.cc

```diff
@@ -341,24 +341,28 @@ seastar::future<> OSD::_add_me_to_crush()
   if (!local_conf().get_val<bool>("osd_crush_update_on_start")) {
     return seastar::now();
   }
-  const double weight = [this] {
+  auto get_weight = [this] () {
     if (auto w = local_conf().get_val<double>("osd_crush_initial_weight");
 	w >= 0) {
-      return w;
+      return seastar::make_ready_future<double>(w);
     } else {
-      auto total = store->stat().total;
-      return std::max(.00001, double(total) / double(1ull << 40)); // TB
+       return store->stat().then([](auto st) {
+         auto total = st.total;
+	 return seastar::make_ready_future<double>(std::max(.00001, double(total) / double(1ull << 40))); // TB
```

Comment:
wrap at 80. and might want to be consistent when using space and tab for indent.